src/modules/load-fields.js

```diff
@@ -0,0 +1,48 @@
+import util from 'util';
+import dotnotation from '../utils/dotnotation';
+
+const DEFAULT_SAMPLE_SIZE = 50;
+const ENABLED = 1;
+
+function extractFieldsFromDocument(doc) {
+  return Object.keys(dotnotation.serialize((doc)));
+}
+
+function truncateFieldToDepth(field, depth) {
+  if (!depth) {
+    return field;
+  }
+
+  return field
+    .split('.')
+    .slice(0, depth)
+    .join('.');
+}
+
+export default async function loadFields(
+  dataService,
+  ns,
+  { filter, sampleSize, maxDepth } = {},
+  driverOptions = {}
+) {
+  const find = util.promisify(dataService.find.bind(dataService));
+
+  const docs = await find(ns, filter || {}, {
+    limit: sampleSize || DEFAULT_SAMPLE_SIZE,
+    ...driverOptions
+  });
```

Comment:
Just for my own understanding, this will mean that if documents that are added to a collection after the first `sampleSize` documents will not affect the result here, and that we are okay with that?

Reply:
Yep, that is a great point.. I don't think we should be. Initially it was based on 1 document find, then on a random sample of 50 docs, which is a bit better but extremely slow on a big collection, causing the field selection dialog to be blank for a while without any indication, that's why we switched it to a 50 docs find.

The only accurate way to get all the fields is a full collection scan, otherwise we need to change approach: have a 'use all the fields option', relay on users building their own projections without the fields selection, let people decide how many document to scan to get the fields or at least indicate fields are being loaded...